openmc/tallies.py

```diff
@@ -1425,25 +1436,47 @@ def export_results(self, filename='tally-results', directory='.',
             # Pickle the Tally results to a file
             pickle.dump(tally_results, open(filename, 'wb'))
 
-    def _outer_product(self, other, binary_op):
+    def hybrid_product(self, other, binary_op, filter_product=None,
+                        nuclide_product=None, score_product=None):
         """Combines filters, scores and nuclides with another tally.
 
-        This is a helper method for the tally arithmetic methods. The filters,
-        scores and nuclides from both tallies are enumerated into all possible
-        combinations and expressed as CrossFilter, CrossScore and
-        CrossNuclide objects in the new derived tally.
+        This is a helper method for the tally arithmetic methods. It is called a
+        "hybrid product" because it performs a combination of tensor
+        (or Kronecker) and entrywise (or Hadamard) products. The filters from
+        both tallies are combined using an entrywise (or Hadamard) product on
+        matching filters. By default, if all nuclides are identical in the two
+        tallies, the entrywise product is performed across nuclides; else the
+        tensor product is performed. By default, if all scores are identical in
+        the two tallies, the entrywise product is performed across scores; else
+        the tensor product is performed. Users can also call the method
```

Comment:
I think you might clarify this comment block to say that it is a helper method for the "tally arithmetic operator overloaded  methods" since you mention later that "user can also call the method explicitly..." For new users/developers it may not be clear that "tally arithmetic" really just means super fancy operator overloading.